src/appleseed/renderer/kernel/shading/closures.h

```diff
@@ -107,6 +107,13 @@ enum ClosureID
     NPRShadingID,
     NPRContourID,
 
+    // Microfacet normal mapping closures.
+    MicrofacetGlossyID,
+    MicrofacetBlinnID,
+    MicrofacetMetalID,
```

Comment:
The Lambertian is missing here too.

Reply:
Explaned here: https://github.com/appleseedhq/appleseed/pull/2886#discussion_r480894409